apps/guide/src/content/03-creating-your-bot/05-registering-slash-commands.mdx

```diff
@@ -0,0 +1,143 @@
+import config from './config.json' assert { type: 'json' };
+
+const { clientId, guildId, token } = config;
+
+const commands = [];
+
+// Grab all the command files from the commands directory you created earlier
+const foldersPath = fileURLToPath(new URL('commands', import.meta.url));
+const commandFolders = await readdir(foldersPath);
+
+for (const folder of commandFolders) {
+	// Grab all the command files from the commands directory you created earlier
+	const commandsPath = join(foldersPath, folder);
+	const commandFiles = await readdir(commandsPath).then((files) => files.filter((file) => file.endsWith('.js')));
+	// Grab the SlashCommandBuilder#toJSON() output of each command's data for deployment
+	for (const file of commandFiles) {
+		const filePath = join(commandsPath, file);
+		const command = await import(filePath);
+		if ('data' in command && 'execute' in command) {
+			commands.push(command.data.toJSON());
+		} else {
+			console.log(`[WARNING] The command at ${filePath} is missing a required "data" or "execute" property.`);
+		}
+	}
+}
+
+// Construct and prepare an instance of the REST module
+const rest = new REST().setToken(token);
+
+try {
+	console.log(`Started refreshing ${commands.length} application (/) commands.`);
+
+	// The put method is used to fully refresh all commands in the guild with the current set
+	const data = await rest.put(Routes.applicationGuildCommands(clientId, guildId), { body: commands });
+
+	console.log(`Successfully reloaded ${data.length} application (/) commands.`);
+} catch (error) {
+	// And of course, make sure you catch and log any errors!
+	console.error(error);
+}
+```
+
+</CH.Code>
+
+Once you fill in these values, run _`node deploy-commands.js`_ in your project directory to register your commands to the guild specified. If you see the success message, check for the commands in the server by typing `/`! If all goes well, you should be able to run them and see your bot's response in Discord!
+
+### Global commands
+
+Global application commands will be available in all the guilds your application has the _`applications.commands`_ scope authorized in, and in direct messages by default.
+
+To deploy global commands, you can use the same script from the [guild commands](#guild-commands) section and simply adjust the route in the script to `.applicationCommands(clientId)`
+
+<CH.Code>
+
+```js focus=2
+await rest.put(Routes.applicationCommands(clientId), { body: commands });
+```
+
+</CH.Code>
+
+### Where to deploy
+
+<Alert title="Tip" type="success">
+Guild-based deployment of commands is best suited for development and testing in your own personal server. Once you're satisfied that it's ready, deploy the command globally to publish it to all guilds that your bot is in.
+
+You may wish to have a separate application and token in the Discord Dev Portal for your dev application, to avoid duplication between your guild-based commands and the global deployment.
+
+</Alert>
```

Comment:
```suggestion
<Alert title="Tip" type="success">
	Guild-based deployment of commands is best suited for development and testing in your own personal server. Once you're satisfied that it's ready, deploy the command globally to publish it to all guilds that your bot is in.

	You may wish to have a separate application and token in the Discord Dev Portal for your dev application, to avoid duplication between your guild-based commands and the global deployment.
</Alert>
```